### DISCUSSION TITLE
Creating table and importing data from CSV

### DISCUSSION THREAD

**Discussion by daviwall** [Budibase/budibase#2653] [Help] [answered]
Hello everyone,

I have installed budibase docker on an AWS instance. I wanted to create a table and import data from a CSV which I exported from an application. When I import the CSV with full data, which is roughly 4500 records, the import functionality just hangs and never finishes. I tried leaving it running for quite a while to no avail. I tried then increase the instance type in AWS to a t2 xlarge and I get the same behavior.
Now if I try to import the "same CSV" but with like 50 records, it works just fine. Has anyone had this issue before? 
Can anyone point me out to the right direction here?

Thank you very much! :)

**Comment by shogunpurple** [comment, marked as answer]
Hi @daviwall - this seems to be related. https://github.com/Budibase/budibase/issues/1840

We will be looking at addressing the performance of CSV imports over the next week or so, and should get this resolved.

**Reply by daviwall** [reply]
Thank you for the information, looking forward for a solution.
Is there another option to import data into budibase?

**Reply by mjashanks** [reply]
Hey @daviwall - There is no other official way at the moment, 

It's possible to push records into Budibase using the REST API, and a script. However, our API is not documented. You would need to 
1. Build your table
2. Run your app, and open your "New Row" screen, for the table.
3. Open developer tools and watch the network tab
4. Fill out the form, and press save. 

In your network tab, you will see a call that looks like `POST api/<table ID>/rows` - you would need to mimic this call, including the cookie (for authorisation) and the JSON body.

If you're at all uncomfortable with that, I'd stay away from it!

**Reply by daviwall** [reply]
Hi Michael, that is beautiful!

I will create a python script to do that!

Thank you very much.
